packages/x-tree-view/src/internals/plugins/useTreeViewItems/useTreeViewItems.tsx

```diff
@@ -181,14 +181,15 @@ export const useTreeViewItems: TreeViewPlugin<UseTreeViewItemsSignature> = ({
   const getItemTree = React.useCallback(() => {
     const getItemFromItemId = (itemId: TreeViewItemId): TreeViewBaseItem => {
       const item = itemsSelectors.itemModel(store.state, itemId);
+      const itemToMutate = { ...item };
```

Comment:
@flaviendelangle This only makes a clone of the outer-most item.

E.g:

```ts
const x = {a: {b: []}};
const copy = {...x};
copy.a.b = ['new'];
console.log(x)
// { a: { b: [ 'new' ] } }
```

You need to use `structuredClone` instead:

```ts
const x = {a: {b: []}};
const copy = structuredClone(x);
copy.a.b = ['new'];
console.log(x)
// { a: { b: [ 'new' ] } }
```